chore(3663): removing the isMultichainVersion1Enabled feature flag, and corresponding dead code

EtherWizard33 opened the pull request:
<!--
Please submit this PR as a draft initially.
Do not mark it as "Ready for review" until the template has been completely filled out, and PR status checks have passed at least once.
-->

## **Description**
- Step 1 **(done)**: remove code that ran in the feature flag OFF scenario, confirm stuff runs as expected (manually and e2e)
- Step 2 **(done)**: remove any left over checks on the feature flags, for examples checks that used only feature flag ON to render something specific to that scenario
- Step 3 **(done)**: address unit test snapshots tests failing, all the while using these snapshots to ensure expected behavior is maintained, comparing a test that runs with FF OFF, to the existing snapshot on main, this test should pass since FF OFF is the default, this a baseline. Next, compare the snapshot generated from a FF ON test running on the code with feature flag, to the snapshot generated once all the dead code and feature flag code is removed, that test should pass since FF ON becomes the default behavior once feature flag and dead code is removed.
- Step 4 **(done)**: address unit tests that fail on things other than snapshots, find out why they are failing
- Step 5 **(done)**: added coverage on new code (Sonar)

<!--
Write a short description of the changes included in this pull request, also include relevant motivation and context. Have in mind the following questions:
1. What is the reason for the change?
3. What is the improvement/solution?
-->

## **Related issues**

Contributes to: https://github.com/MetaMask/MetaMask-planning/issues/3663

## **Manual testing steps**

1. Go to the in app browser and test-dapp
2. Connect to it
3. Click the top header in the in-app browser, confirm the expected permissions as displaying.

## **Screenshots/Recordings**

<!-- If applicable, add screenshots and/or recordings to visualize the before and after of your change. -->

### **Before**

It's a refactoring, to the user it looks the same before and after.

### **After**

<!-- [screenshots/recordings] -->

## **Pre-merge author checklist**

- [ ] I’ve followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Mobile Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

EtherWizard33 committed:
chore: toggling the chain permissions feature flags so they are always true, to check CI

EtherWizard33 committed:
chore: remove MC FF OFF code (dead code) from NetworkVerificationInfo component

EtherWizard33 committed:
chore: remove MC FF OFF code (dead code) from AccountConnectMultiSelector component

EtherWizard33 committed:
chore: remove MC FF OFF code (dead code) from AccountPermissions component

EtherWizard33 committed:
chore: remove MC FF OFF code (dead code) from AccountPermissionsConnected component

EtherWizard33 committed:
chore: remove MC FF OFF code (dead code) from NetworkPermissionsConnected component

EtherWizard33 committed:
chore: remove MC FF ON check and when applicatable FF OFF Code (dead code) from NetworkVerificationInfo component

EtherWizard33 committed:
refactor(3663): AccountConnect remove checks on feature flag, remove code that ran when feature flag was off (which was now dead code)

EtherWizard33 committed:
refactor(3663): for AccountPermissions component remove checks on feature flag, remove code that ran when feature flag was off (which was now dead code)

EtherWizard33 committed:
fix: previous commit on this AccountConnect file introduced bugs, this is now fixed

EtherWizard33 committed:
refactor(3663): for AccountConnectMultiSelector component remove checks on feature flag, remove code that ran when feature flag was off (which was now dead code)

EtherWizard33 committed:
chore(test): remove feature flag-off code, since it is to be dead code. Update the test to mock nav, and snapshot expected to be updated too.

EtherWizard33 committed:
style: lint

EtherWizard33 committed:
chore(test): for NetworkVerification component, update existing unit tests.

In more detail, after removal of feature flag and corresponding dead code of ff-off, test now runs on ff-on and required update to show content of accordion before asserting, and update snapshot to match new expected rendering.

EtherWizard33 committed:
Merge branch 'main' into refactor-3663-chain-permissions-feature-flag-etc

EtherWizard33 committed:
test: updating snapshot of account permissions since button label text component default style was changed in main

EtherWizard33 committed:
chore(test): update snapshot after having removed the feature flag for multichain and the deadcode behind it.

In the case of this component, it didn't directly have multichain code in it but was rendering a component which did (NetworkVerificationInfo)

EtherWizard33 committed:
chore(test): update NetworkSwitcher snapshot after having removed the feature flag for multichain and the deadcode behind it.

It needs this since it uses components than end up using NetworkVerificationInfo which had some code toggled on and off based on the multichain feature flag which is being removed in this PR.

EtherWizard33 committed:
chore: remove the feature flag check that checked if the multichain feature flag was ON.

This check on the feature flag was used in some components to render stuff that should render only when the flag was ON.

EtherWizard33 committed:
style: lint

EtherWizard33 committed:
chore(test): remove backup snapshots that were used for temporary troubleshooting during code cleanup

EtherWizard33 committed:
style: tsc lint

EtherWizard33 committed:
Merge branch 'temp' into ttemp

EtherWizard33 committed:
chore(test): update snapshot to match expected and cleanup ff-on test since all multichain-permission code is ff-on now

EtherWizard33 committed:
test: add test case that covers when the permission summary is rendered

EtherWizard33 committed:
Merge branch 'main' into refactor-3663-chain-permissions-feature-flag-etc

EtherWizard33 committed:
test: covering both cases of sdk url known and unknown

tommasini commented on app/components/Views/AccountPermissions/AccountPermissions.test.tsx:
```diff
@@ -27,11 +27,11 @@ jest.mock('../../../components/hooks/useMetrics', () => ({
-  isMultichainVersion1Enabled: false,
-  isChainPermissionsFeatureEnabled: false,
-}));
+// jest.mock('../../../util/networks/index.js', () => ({
```
Can this code be removed?

EtherWizard33 replied:
yes, thanks for noticing, code is now removed

tommasini commented on app/util/networks/index.js:
```diff
@@ -488,11 +488,9 @@ export const getBlockExplorerTxUrl = (
 
-export const isMultichainVersion1Enabled =
-  process.env.MM_MULTICHAIN_V1_ENABLED === 'true';
+export const isMultichainVersion1Enabled = true;
```
Is this dead code now as well? Or still missing a few places?

EtherWizard33 replied:
I have now removed isMultichainVersion1Enabled = true; 

For the other one, isChainPermissionsFeatureEnabled = true, I wrote to Alex Donetsky since it's was his.
I think its better that he remove that other feature flag, so that this PR stays in the scope it was intended.

EtherWizard33 committed:
test: Add coverage of the confirmation click handler

EtherWizard33 committed:
Merge branch 'main' into refactor-3663-chain-permissions-feature-flag-etc

EtherWizard33 committed:
test(account-connect): update the snapshot capturing the bottom sheet displayed at initial connection by a dapp.

Main branch has an updted version of this component, that, at the time of initial connection, pre-selects all networks. This is the change reflected in this updated snapshot.

EtherWizard33 committed:
test(account-permissions): update the snapshot for the AccountPermissions component.

Creating a snapshot on current main with ff-on, matches the expected snapshot when running the test on the feature branch, which has the feature flag and dead code removed. This proves the result in this feature branch is the same currently in prod. Therefore removing the feature flag does not change this component in prod, at least for the state the snapshot captures.

EtherWizard33 committed:
test(snapshot): in the previous merge commit, git seems to have had trouble identifying merge conflicts in snapshot file, had to ensure all was as expected and regenerate it.

This branch removing all ff-off code, the expected snapshot should be the same as running main with ff-on. So we generate the snapshot on main, overwrite the one in the feaure branch then run the test to ensure the code without feature flag generates the same snapshot as the one with ff-ON on main.

EtherWizard33 committed:
test(snapshot): for NetworkSwitcher, in the previous merge commit, git seems to have had trouble identifying merge conflicts in snapshot file, had to ensure all was as expected and regenerate it.

This branch is removing all ff-off code, the expected snapshot should be the same as running main with ff-on. So we generate the snapshot on main, overwrite the one in the feaure branch then run the test to ensure the code without feature flag generates the same snapshot
 as the one with ff-ON on main.

EtherWizard33 committed:
test(account-permissions-confirm-revoke-all): Add unit tests, the coverage for the component went from 53% to 80%.

EtherWizard33 committed:
Merge branch 'main' into refactor-3663-chain-permissions-feature-flag-etc

EtherWizard33 committed:
test(network-modal): Update snapshot.

The this snapshot matches exactly the snapshot when main runs with multichain feature flag on. And so does the code in the feature branch its being commited to, since, as expected. This branch removed feature flag off code, keeping only ff ON code.

EtherWizard33 committed:
test(switch-custom-network): Update snapshot of the SwitchCustomNetwork compoennt.

    The this snapshot matches exactly the snapshot when main runs with multichain feature flag on. And so does the code in the feature branch its being commited to, since, as expected. This branch removed feature flag off code, keeping only ff ON code.

EtherWizard33 committed:
test(cleanup): remove commented code

EtherWizard33 committed:
test(cleanup): Removing deprecated test.

This test is now deprecated. It was meant to test the code when multichain permissions feature flag was off, in a seperate file, in order to allow mocking of the feature flag getters. Since the code has now removed the feautre flag, all tests on multichain permission features now run in ff-on mode.

EtherWizard33 committed:
chore(feature-flag): Remove feature flag from browser tab component.

The multichain permissions feature flag is now removed accross the app. Therefore the browser tab always runs in ff-on mode going forward.

EtherWizard33 committed:
chore(feature-flag): Remove the getter that was use to access the chain permission UI feature flag.

EtherWizard33 committed:
Merge branch 'main' into
refactor-3663-chain-permissions-feature-flag-etc

EtherWizard33 committed:
test: temp comment

EtherWizard33 committed:
Revert "test: temp comment"

This reverts commit 2729052ee6569c1f8f4f35c9f1fae1d4b187ba0c.

EtherWizard33 committed:
test(AccountPermissions): add test for manage permissions navigation

Add test case to verify navigation to permissions summary when manage permissions button is pressed in AccountPermissions component

EtherWizard33 committed:
test(Settings): add test for permissions settings visibility

Add test case to verify permissions settings button renders when feature flag is enabled

wachunei reviewed (approved):
Ramp changes LGTM

vinnyhoward reviewed (approved):
LGTM

EtherWizard33 committed:
Merge branch 'main' into refactor-3663-chain-permissions-feature-flag-etc
